### PR TITLE
use MAX_JSON_TOKEN_COUNT in parsing sdp json

### PR DESCRIPTION
**Pull request by chehefen**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,6 +20,7 @@ install:
   - sudo apt-get install -y gdb
 
 script:
+  - export AWS_KVS_LOG_LEVEL=2
```
Comment (Sean-Der) [inline .travis.yml]: Do we need this?

Comment (MushMal) [reply .travis.yml]: This is the debug level. We can get some info from the runs.

File: src/source/PeerConnection/SessionDescription.h [hunk 2]
```diff
@@ -10,8 +10,8 @@ SessionDescription internal include file
 extern "C" {
 #endif
 
-#define SESSION_DESCRIPTION_TOKENS 5
-#define ICE_CANDIDATE_INIT_TOKENS 7
+// For tight packing
+#pragma pack(push, include_i, 1) // for byte alignment
```
Comment (MushMal) [inline src/source/PeerConnection/SessionDescription.h]: Remove these - we don't tight pack any longer or merge from mainline

File: tst/PeerConnectionFunctionalityTest.cpp [hunk 3]
```diff
@@ -19,17 +34,28 @@ TEST_F(PeerConnectionFunctionalityTest, connectTwoPeers)
...
 
     auto onICECandidateHdlr = [](UINT64 customData, PCHAR candidateStr) -> void {
-        RtcIceCandidateInit iceCandidate;
+        PAddIceCandidateRoutineArgs pAddIceCandidateRoutineArgs = NULL;
+        TID tid;
 
         if (candidateStr != NULL) {
-            EXPECT_EQ(deserializeRtcIceCandidateInit(candidateStr, STRLEN(candidateStr), &iceCandidate), STATUS_SUCCESS);
-            EXPECT_EQ(addIceCandidate((PRtcPeerConnection) customData, iceCandidate.candidate), STATUS_SUCCESS);
+            pAddIceCandidateRoutineArgs = (PAddIceCandidateRoutineArgs) MEMALLOC(SIZEOF(AddIceCandidateRoutineArgs));
```
Comment (MushMal) [inline tst/PeerConnectionFunctionalityTest.cpp]: Doesn't need to be allocated/freed - can be on the stack variable

**Comment by chehefen** [comment]
> Thank you so much for digging into this! How did you find/diagnose the deadlock? Would love to know what helped/what can be improved debugging wise.

I couldnt pass the travis checks so I run the test locally myself and found the deadlock.

**Pull request merged**